components/brave_vpn/brave_vpn_service.cc

```diff
@@ -865,9 +920,10 @@ void BraveVpnService::GetSubscriberCredential(
 }
 
 void BraveVpnService::GetSubscriberCredentialV12(ResponseCallback callback) {
-  api_request_.GetSubscriberCredentialV12(
-      std::move(callback), skus_credential_,
-      GetBraveVPNPaymentsEnv(GetCurrentEnvironment()));
+  // Caller can get valid subscriber credential only in purchased state.
+  // Otherwise, false is passed to |callback| as success param.
+  std::move(callback).Run(::brave_vpn::GetSubscriberCredential(local_prefs_),
```

Comment:
@deeppandya @bsclifton , As we cache subscriber credential in local state, we can return subscriber credential to caller. Previously, we fetch subscriber credential whenever this method is called with stored `skus_credential_`.
Caller will got `false` as a `success` param when stored subscriber credential is already expired.

Reply:
@deeppandya I think we don't need to worry about expiration with this background refreshing -  https://github.com/brave/brave-core/pull/15702/commits/5c17a6c0ab3d280defeeb13d1f0e80604499bfda